src/Resources/contao/drivers/DC_Table.php

```diff
@@ -6145,8 +6145,9 @@ public function initPicker(PickerInterface $picker)
 			// Allow only those roots that are allowed in root nodes
 			if (!empty($this->root) && $arrRoot != $this->root)
 			{
+				$arrRoot = array_merge($arrRoot, $this->Database->getChildRecords($arrRoot, $this->strTable));
 				$arrRoot = array_intersect($arrRoot, array_merge($this->root, $this->Database->getChildRecords($this->root, $this->strTable)));
-				$arrRoot = $this->eliminateNestedPages($arrRoot);
+				$arrRoot = $this->eliminateNestedPages(array_unique($arrRoot, SORT_NUMERIC));
```

Comment:
Why did you add the `SORT_NUMERIC` flag?

Reply:
> Why did you add the SORT_NUMERIC flag?

To compare the array values numerically.

But your solution in f3aafc07a0d830900e2fdccccd22048991471a23 works, too.